### PR TITLE
Source metabase: allow using only https urls

### LINKED ISSUE #16519: Add check to prevent users from using http url with Metabase
Tell us about the problem you're trying to solve
Currently, it is possible to use unsecured URL (http) in the Metabase config.

Describe the solution you’d like
We should fail the CHECK if a user enters a non-https URL for Metabase.

### PR DESCRIPTION
**Pull request by davydov-d**
## What
Disallow using `http` URLs

## How
Using pattern in a spec

### CODE REVIEW

File: airbyte-integrations/connectors/source-metabase/source_metabase/spec.yaml [hunk 1]
```diff
@@ -14,6 +14,7 @@ connectionSpecification:
         URL to your metabase instance API
       examples:
         - "http://localhost:3000/api/"
+      pattern: "https://.+"
```
Comment (grubberr) [inline airbyte-integrations/connectors/source-metabase/source_metabase/spec.yaml:17]: I can bypass your regex for example 'hhhhttps://google.com' :)

Comment (grubberr) [reply airbyte-integrations/connectors/source-metabase/source_metabase/spec.yaml:17]: how about "^https://" ?

Comment (davydov-d) [reply airbyte-integrations/connectors/source-metabase/source_metabase/spec.yaml]: good catch, thanks! done

**Review by grubberr — APPROVED** [review]

**Pull request merged**